### PR TITLE
CB-9243 AWS PlacementGroup Support

### PR DESCRIPTION
**Pull request by smaniraju**
Placement Group Configuration support for AWS Clusters via Cluster Definitions. The PlacementGroup configuration for AWS is done similar to existing AvailabilitySet Configuration for Azure Clusters.

1. Tested multiple DH Cluster launches with PG strategy as NONE, SPREAD, CLUSTER, PARTITION.
2. Tested multiple DH Cluster launches without any PG Configuration.
3. Tested default DH templates launched successfully and were configured with correct PG when PG Config was added.
4. Tested successful AWS Environment\DL launches without any PG Configuration.

### CODE REVIEW

**Comment by smaniraju** [comment]
Default cluster definitions update to configure the appropriate  PG strategy would be taken by template owning teams like COD team for aws and also appropriate AvailabilitySets in default azure cluster definitions.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/parameter/template/AwsInstanceTemplateV4Parameters.java
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/parameter/template/AwsInstanceTemplateV4Parameters.java:120]: else branch should be great here which maps to none

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/parameter/template/AwsInstanceTemplateV4Parameters.java
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/parameter/template/AwsInstanceTemplateV4Parameters.java:78]: if null put none into the map

File: cloud-aws/src/main/resources/templates/aws-cf-stack.ftl
Comment (doktoric) [inline cloud-aws/src/main/resources/templates/aws-cf-stack.ftl:394]: why we dont put lower in the java code side ?

File: cloud-aws/src/main/resources/templates/aws-cf-stack.ftl
Comment (doktoric) [inline cloud-aws/src/main/resources/templates/aws-cf-stack.ftl:372]: use placementgroupname  here and generate in java code

**Review by doktoric — APPROVED** [review]

**Comment by doktoric** [comment]
@lacikaaa can we get a second review on this?

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**
